src/Microsoft.Framework.DependencyInjection/ServiceLookup/ServiceTable.cs

```diff
@@ -47,6 +59,19 @@ public ServiceTable(IEnumerable<ServiceDescriptor> descriptors)
                 }
                 else
                 {
+                    Debug.Assert(descriptor.ImplementationType != null);
+                    var implementationTypeInfo = descriptor.ImplementationType.GetTypeInfo();
```

Comment:
cc @Eilon. Rebasing on dev was causing one of the more recently added tests to fail. The issue was that we were attempting to initialize one of the constructors for `List<>` which then blew up with an exception about being unable to create a void type array. It seems simpler to perform a pre-check as part of registration so we never have to deal with these cases later on.